packages/react-router-dom/index.tsx

```diff
@@ -514,10 +576,10 @@ export function RouterProvider({
   // When we start a view transition, create a Deferred we can use for the
   // eventual "completed" render
   React.useEffect(() => {
-    if (vtContext.isTransitioning) {
+    if (vtContext.isTransitioning && !vtContext.flushSync) {
```

Comment:
This prevents us from ever kicking off the effect chain for `flushSync` view transitions.  No `renderDfd` means the rest of the effects will no-op